### PR TITLE
Client Landing Page and My account method and Transaction Method

### LINKED ISSUE #9: Client Landinging Page and Client Contoller

### PR DESCRIPTION
**Pull request by samkelisojam**

### CODE REVIEW

File: 2024Project-main/2024FinalYearProject/Controllers/ClientController.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        //         Transactions = transactions
+        //     };
+        //     return View(viewModel);
+        // }
+
+        [HttpGet]
+        public async Task<IActionResult> CashSent()
+        {
+            var username = User.Identity.Name;
+
```
Comment (ghost) [inline 2024Project-main/2024FinalYearProject/Controllers/ClientController.cs:70]: remove unnecessary spaces.(have one space between lines)

File: 2024Project-main/2024FinalYearProject/Controllers/ClientController.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            return View(model);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CashSent(CashSentViewModel model)
+        {
+            var username = User.Identity.Name;
+
+
```
Comment (ghost) [inline 2024Project-main/2024FinalYearProject/Controllers/ClientController.cs]: spaces

File: 2024Project-main/2024FinalYearProject/Controllers/ClientController.cs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                Reference = "CashSent",
+            };
+
+            // Generate random Cash Digit and PIN
+            var cashDigit = GenerateRandomNumber(13);
+            var pinNumber = GenerateRandomNumber(4);
+
+            // Save the transaction
+            await _repo.Transaction.AddAsync(transaction);
+           await _repo.Transaction.SaveAsync();
```
Comment (ghost) [inline 2024Project-main/2024FinalYearProject/Controllers/ClientController.cs]: remove this unnecessary SaveAsync , it is already handled on the base

File: 2024Project-main/2024FinalYearProject/Data/Interfaces/IRepositoryBase.cs [hunk 4]
```diff
@@ -7,5 +7,6 @@ public interface IRepositoryBase<T> where T : class
         Task<T> GetByIdAsync(int? id);
         Task RemoveAsync(int? id);
         Task UpdateAsync(T? entity);
+        Task SaveAsync();
```
Comment (ghost) [inline 2024Project-main/2024FinalYearProject/Data/Interfaces/IRepositoryBase.cs]: This is unnecessary (must be removed)

File: 2024Project-main/2024FinalYearProject/Data/RepositoryBase.cs [hunk 5]
```diff
@@ -22,7 +22,11 @@ public async Task<List<T>> GetAllAsync()
         {
             return await _context.Set<T>().ToListAsync();
         }
-
+        public async Task<string> GetUsernameByIdAsync(string userId)
+        {
```
Comment (ghost) [inline 2024Project-main/2024FinalYearProject/Data/RepositoryBase.cs:26]: This is a duplicate of GetByIdAsync , you just need to add IUserRepository or IAccountRepository to the wrapper and files

File: 2024Project-main/2024FinalYearProject/Data/RepositoryBase.cs [hunk 6]
```diff
@@ -44,5 +48,10 @@ public async Task UpdateAsync(T? entity)
             await _context.SaveChangesAsync();
 
         }
+
+        public async Task SaveAsync()
```
Comment (ghost) [inline 2024Project-main/2024FinalYearProject/Data/RepositoryBase.cs]: Remove this unnecessary method , SaveAsync already handled on AddAsync and UpdateAsync (see line 48)

**Comment by samkelisojam** [comment]
Conflict resolved

File: 2024Project-main/2024FinalYearProject/2024FinalYearProject.csproj [hunk 7]
```diff
@@ -8,35 +8,8 @@
...
-    <None Include="wwwroot\lib\bootstrap\dist\js\bootstrap.esm.js.map" />
-    <None Include="wwwroot\lib\bootstrap\dist\js\bootstrap.esm.min.js" />
-    <None Include="wwwroot\lib\bootstrap\dist\js\bootstrap.esm.min.js.map" />
-    <None Include="wwwroot\lib\bootstrap\dist\js\bootstrap.js" />
-    <None Include="wwwroot\lib\bootstrap\dist\js\bootstrap.js.map" />
-    <None Include="wwwroot\lib\bootstrap\dist\js\bootstrap.min.js" />
-    <None Include="wwwroot\lib\bootstrap\dist\js\bootstrap.min.js.map" />
-    <None Include="wwwroot\lib\bootstrap\LICENSE" />
+    <None Include="Views\Client\Index.cshtml" />
+    <None Include="Views\Client\MyAccount.cshtml" />
```
Comment (ghost) [inline 2024Project-main/2024FinalYearProject/2024FinalYearProject.csproj:12]: is it necessary for these to be added here

File: 2024Project-main/2024FinalYearProject/Data/RepositoryBase.cs (hunk 5 above)
Comment (samkelisojam) [reply 2024Project-main/2024FinalYearProject/Data/RepositoryBase.cs:26]: we will add IAccountRepository along way during the way 
above methodhas been eliminated

**Pull request merged**